### ISSUE TITLE
ClassNotFound exception on httpcomponents library in native mode

### ISSUE THREAD

**Issue by jasmdk** [quarkiverse/quarkus-opensearch#257]
Using quarkus-opensearch-java-client in version 1.8.1 I get a ClassNotFoundException at runtime when running in native mode (java.lang.ClassNotFoundException: org.apache.hc.client5.http.impl.auth.BasicScheme).
I noticed that both httpcomponents version 4 and 5 was resolved as a dependency, which can be a problem in itself, but excluding version 4 did not fix the issue.
```
{
    "timestamp": "2024-06-13 07:04:20",
    "sequence": 1419,
    "loggerClassName": "org.slf4j.impl.Slf4jLogger",
    "loggerName": "org.apache.hc.client5.http.impl.auth.BasicAuthCache",
    "level": "WARN",
    "message": "Unexpected error while de-serializing auth scheme",
    "threadName": "executor-thread-2",
    "threadId": 49,
    "mdc": {},
    "ndc": "",
    "hostName": "d74bd72d05aa",
    "processId": 54,
    "exception": {
        "refId": 1,
        "exceptionType": "java.lang.ClassNotFoundException",
        "message": "org.apache.hc.client5.http.impl.auth.BasicScheme",
        "frames": [
            {
                "class": "com.oracle.svm.core.hub.ClassForNameSupport",
                "method": "forName",
                "line": 122
            },
            {
                "class": "com.oracle.svm.core.hub.ClassForNameSupport",
                "method": "forName",
                "line": 86
            },
            {
                "class": "java.lang.Class",
                "method": "forName",
                "line": 1359
            },
            {
                "class": "java.lang.Class",
                "method": "forName",
                "line": 1348
            },
            {
                "class": "java.io.ObjectInputStream",
                "method": "resolveClass",
                "line": 804
            },
            {
                "class": "java.io.ObjectInputStream",
                "method": "readNonProxyDesc",
                "line": 2061
            },
            {
                "class": "java.io.ObjectInputStream",
                "method": "readClassDesc",
                "line": 1927
            },
            {
                "class": "java.io.ObjectInputStream",
                "method": "readOrdinaryObject",
                "line": 2252
            },
            {
                "class": "java.io.ObjectInputStream",
                "method": "readObject0",
                "line": 1762
            },
            {
                "class": "java.io.ObjectInputStream",
                "method": "readObject",
                "line": 540
            },
            {
                "class": "java.io.ObjectInputStream",
                "method": "readObject",
                "line": 498
            },
            {
                "class": "org.apache.hc.client5.http.impl.auth.BasicAuthCache",
                "method": "get",
                "line": 196
            },
            {
                "class": "org.apache.hc.client5.http.impl.auth.BasicAuthCache",
                "method": "get",
                "line": 155
            },
            {
                "class": "org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport$WrappingAuthCache",
                "method": "get",
                "line": 860
            },
            {
                "class": "org.apache.hc.client5.http.auth.AuthCache",
                "method": "get",
                "line": 91
            },
            {
                "class": "org.apache.hc.client5.http.impl.auth.AuthCacheKeeper",
                "method": "loadFromCache",
                "line": 106
            },
            {
                "class": "org.apache.hc.client5.http.impl.auth.AuthCacheKeeper",
                "method": "loadPreemptively",
                "line": 93
            },
            {
                "class": "org.apache.hc.client5.http.impl.async.AsyncProtocolExec",
                "method": "execute",
                "line": 165
            },
            {
                "class": "org.apache.hc.client5.http.impl.async.AsyncExecChainElement",
                "method": "execute",
                "line": 54
            },
            {
                "class": "org.apache.hc.client5.http.impl.async.AsyncRedirectExec",
                "method": "internalExecute",
                "line": 113
            },
            {
                "class": "org.apache.hc.client5.http.impl.async.AsyncRedirectExec",
                "method": "execute",
                "line": 281
            },
            {
                "class": "org.apache.hc.client5.http.impl.async.AsyncExecChainElement",
                "method": "execute",
                "line": 54
            },
            {
                "class": "org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient",
                "method": "executeImmediate",
                "line": 372
            },
            {
                "class": "org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient",
                "method": "lambda$doExecute$0",
                "line": 228
            },
            {
                "class": "org.apache.hc.core5.http.nio.support.BasicRequestProducer",
                "method": "sendRequest",
                "line": 93
            },
            {
                "class": "org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient",
                "method": "doExecute",
                "line": 203
            },
            {
                "class": "org.apache.hc.client5.http.impl.async.CloseableHttpAsyncClient",
                "method": "execute",
                "line": 97
            },
            {
                "class": "org.apache.hc.client5.http.impl.async.CloseableHttpAsyncClient",
                "method": "execute",
                "line": 107
            },
            {
                "class": "org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport",
                "method": "performRequestAsync",
                "line": 212
            },
            {
                "class": "org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport",
                "method": "performRequestAsync",
                "line": 175
            },
            {
                "class": "org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport",
                "method": "performRequest",
                "line": 148
            },
            {
                "class": "org.opensearch.client.opensearch.cluster.OpenSearchClusterClient",
                "method": "health",
                "line": 360
            },
            {
                "class": "io.quarkiverse.opensearch.client.runtime.health.OpenSearchHealthCheck",
                "method": "call",
                "line": 26
            },
            {
                "class": "io.quarkiverse.opensearch.client.runtime.health.OpenSearchHealthCheck_ClientProxy",
                "method": "call"
            },
            {
                "class": "io.smallrye.context.impl.wrappers.SlowContextualSupplier",
                "method": "get",
                "line": 21
            },
            {
                "class": "io.smallrye.mutiny.operators.uni.builders.UniCreateFromItemSupplier",
                "method": "subscribe",
                "line": 28
            },
            {
                "class": "io.smallrye.mutiny.operators.AbstractUni",
                "method": "subscribe",
                "line": 36
            },
            {
                "class": "io.smallrye.mutiny.operators.uni.UniOnFailureFlatMap",
                "method": "subscribe",
                "line": 31
            },
            {
                "class": "io.smallrye.mutiny.operators.AbstractUni",
                "method": "subscribe",
                "line": 36
            },
            {
                "class": "io.smallrye.mutiny.operators.uni.UniOnItemTransform",
                "method": "subscribe",
                "line": 22
            },
            {
                "class": "io.smallrye.mutiny.operators.AbstractUni",
                "method": "subscribe",
                "line": 36
            },
            {
                "class": "io.smallrye.mutiny.operators.uni.UniRunSubscribeOn",
                "method": "lambda$subscribe$0",
                "line": 27
            },
            {
                "class": "io.smallrye.mutiny.vertx.MutinyHelper",
                "method": "lambda$blockingExecutor$6",
                "line": 62
            },
            {
                "class": "io.vertx.core.impl.ContextImpl",
                "method": "lambda$executeBlocking$1",
                "line": 191
            },
            {
                "class": "io.vertx.core.impl.ContextInternal",
                "method": "dispatch",
                "line": 279
            },
            {
                "class": "io.vertx.core.impl.ContextImpl",
                "method": "lambda$internalExecuteBlocking$2",
                "line": 210
            },
            {
                "class": "io.quarkus.vertx.core.runtime.VertxCoreRecorder$14",
                "method": "runWith",
                "line": 599
            },
            {
                "class": "org.jboss.threads.EnhancedQueueExecutor$Task",
                "method": "doRunWith",
                "line": 2516
            },
            {
                "class": "org.jboss.threads.EnhancedQueueExecutor$Task",
                "method": "run",
                "line": 2495
            },
            {
                "class": "org.jboss.threads.EnhancedQueueExecutor$ThreadBody",
                "method": "run",
                "line": 1521
            },
            {
                "class": "org.jboss.threads.DelegatingRunnable",
                "method": "run",
                "line": 11
            },
            {
                "class": "org.jboss.threads.ThreadLocalResettingRunnable",
                "method": "run",
                "line": 11
            },
            {
                "class": "io.netty.util.concurrent.FastThreadLocalRunnable",
                "method": "run",
                "line": 30
            },
            {
                "class": "java.lang.Thread",
                "method": "runWith",
                "line": 1596
            },
            {
                "class": "java.lang.Thread",
                "method": "run",
                "line": 1583
            },
            {
                "class": "com.oracle.svm.core.thread.PlatformThreads",
                "method": "threadStartRoutine",
                "line": 896
            },
            {
                "class": "com.oracle.svm.core.thread.PlatformThreads",
                "method": "threadStartRoutine",
                "line": 872
            }
        ]
    },
    "stackTrace": ": java.lang.ClassNotFoundException: org.apache.hc.client5.http.impl.auth.BasicScheme\n\tat org.graalvm.nativeimage.builder/com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:122)\n\tat org.graalvm.nativeimage.builder/com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:86)\n\tat java.base@21.0.3/java.lang.Class.forName(DynamicHub.java:1359)\n\tat java.base@21.0.3/java.lang.Class.forName(DynamicHub.java:1348)\n\tat java.base@21.0.3/java.io.ObjectInputStream.resolveClass(ObjectInputStream.java:804)\n\tat java.base@21.0.3/java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:2061)\n\tat java.base@21.0.3/java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1927)\n\tat java.base@21.0.3/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2252)\n\tat java.base@21.0.3/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1762)\n\tat java.base@21.0.3/java.io.ObjectInputStream.readObject(ObjectInputStream.java:540)\n\tat java.base@21.0.3/java.io.ObjectInputStream.readObject(ObjectInputStream.java:498)\n\tat org.apache.hc.client5.http.impl.auth.BasicAuthCache.get(BasicAuthCache.java:196)\n\tat org.apache.hc.client5.http.impl.auth.BasicAuthCache.get(BasicAuthCache.java:155)\n\tat org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport$WrappingAuthCache.get(ApacheHttpClient5Transport.java:860)\n\tat org.apache.hc.client5.http.auth.AuthCache.get(AuthCache.java:91)\n\tat org.apache.hc.client5.http.impl.auth.AuthCacheKeeper.loadFromCache(AuthCacheKeeper.java:106)\n\tat org.apache.hc.client5.http.impl.auth.AuthCacheKeeper.loadPreemptively(AuthCacheKeeper.java:93)\n\tat org.apache.hc.client5.http.impl.async.AsyncProtocolExec.execute(AsyncProtocolExec.java:165)\n\tat org.apache.hc.client5.http.impl.async.AsyncExecChainElement.execute(AsyncExecChainElement.java:54)\n\tat org.apache.hc.client5.http.impl.async.AsyncRedirectExec.internalExecute(AsyncRedirectExec.java:113)\n\tat org.apache.hc.client5.http.impl.async.AsyncRedirectExec.execute(AsyncRedirectExec.java:281)\n\tat org.apache.hc.client5.http.impl.async.AsyncExecChainElement.execute(AsyncExecChainElement.java:54)\n\tat org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient.executeImmediate(InternalAbstractHttpAsyncClient.java:372)\n\tat org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient.lambda$doExecute$0(InternalAbstractHttpAsyncClient.java:228)\n\tat org.apache.hc.core5.http.nio.support.BasicRequestProducer.sendRequest(BasicRequestProducer.java:93)\n\tat org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient.doExecute(InternalAbstractHttpAsyncClient.java:203)\n\tat org.apache.hc.client5.http.impl.async.CloseableHttpAsyncClient.execute(CloseableHttpAsyncClient.java:97)\n\tat org.apache.hc.client5.http.impl.async.CloseableHttpAsyncClient.execute(CloseableHttpAsyncClient.java:107)\n\tat org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.performRequestAsync(ApacheHttpClient5Transport.java:212)\n\tat org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.performRequestAsync(ApacheHttpClient5Transport.java:175)\n\tat org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.performRequest(ApacheHttpClient5Transport.java:148)\n\tat org.opensearch.client.opensearch.cluster.OpenSearchClusterClient.health(OpenSearchClusterClient.java:360)\n\tat io.quarkiverse.opensearch.client.runtime.health.OpenSearchHealthCheck.call(OpenSearchHealthCheck.java:26)\n\tat io.quarkiverse.opensearch.client.runtime.health.OpenSearchHealthCheck_ClientProxy.call(Unknown Source)\n\tat io.smallrye.context.impl.wrappers.SlowContextualSupplier.get(SlowContextualSupplier.java:21)\n\tat io.smallrye.mutiny.operators.uni.builders.UniCreateFromItemSupplier.subscribe(UniCreateFromItemSupplier.java:28)\n\tat io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)\n\tat io.smallrye.mutiny.operators.uni.UniOnFailureFlatMap.subscribe(UniOnFailureFlatMap.java:31)\n\tat io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)\n\tat io.smallrye.mutiny.operators.uni.UniOnItemTransform.subscribe(UniOnItemTransform.java:22)\n\tat io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)\n\tat io.smallrye.mutiny.operators.uni.UniRunSubscribeOn.lambda$subscribe$0(UniRunSubscribeOn.java:27)\n\tat io.smallrye.mutiny.vertx.MutinyHelper.lambda$blockingExecutor$6(MutinyHelper.java:62)\n\tat io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:191)\n\tat io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:279)\n\tat io.vertx.core.impl.ContextImpl.lambda$internalExecuteBlocking$2(ContextImpl.java:210)\n\tat io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:599)\n\tat org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)\n\tat org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)\n\tat org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)\n\tat org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)\n\tat org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)n\tat io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)\n\tat java.base@21.0.3/java.lang.Thread.runWith(Thread.java:1596)\n\tat java.base@21.0.3/java.lang.Thread.run(Thread.java:1583)\n\tat org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:896)\n\tat org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:872)\n"
}
```

**Comment by sboeckelmann** [comment]
@jasmdk can you provide an example project? That will help me a lot to track down the issue.
Thanks!

**Comment by sboeckelmann** [comment]
should be fixed in Release 1.9.6
